### PR TITLE
Move acquireToken code into 'InteractionClients'

### PR DESCRIPTION
**Pull request by tnorling**
Moves the bulk of the acquireToken code into individual InteractionClient classes, scoped to a single request.

### CODE REVIEW

**Review by hectormmg — APPROVED** [review]
Lgtm

**Review by jo-arroyo — APPROVED** [review]

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 1]
```diff
@@ -4,30 +4,26 @@
...
 import { EndSessionRequest } from "../request/EndSessionRequest";
 import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
-import { PopupUtils } from "../utils/PopupUtils";
 import { EndSessionPopupRequest } from "../request/EndSessionPopupRequest";
 import { INavigationClient } from "../navigation/INavigationClient";
-import { NavigationOptions } from "../navigation/NavigationOptions";
 import { EventHandler } from "../event/EventHandler";
+import { PopupClient } from "../interaction_client/PopupClient";
+import { RedirectClient } from "../interaction_client/RedirectClient";
+import { SilentIframeClient } from "../interaction_client/SilentIframeClient";
```
Comment (sameerag) [inline lib/msal-browser/src/app/ClientApplication.ts:25]: `SilentIframe` and `SilentRefresh` - aren't they non-interactive by definition? Will this cause confusion?

Comment (tnorling) [reply lib/msal-browser/src/app/ClientApplication.ts:25]: Open to other suggestions. This was what we came up with during design, keeping in mind the precedence set by the existing `SilentHandler` which extends `InteractionHandler` as well

Comment (sameerag) [reply lib/msal-browser/src/app/ClientApplication.ts:25]: I am thinking, can we have a separate module for silent cases:
`"../silent_client/SilentIframeClient"`;

Comment (pkanher617) [reply lib/msal-browser/src/app/ClientApplication.ts:25]: I think that's just over-optimization, should be fine to leave it as interaction_client.

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 2]
```diff
@@ -417,83 +216,25 @@ export abstract class ClientApplication {
...
-            const state = this.validateAndExtractStateFromHash(hash, InteractionType.Popup, validRequest.correlationId);
-
-            // Remove throttle if it exists
-            ThrottlingUtils.removeThrottle(this.browserStorage, this.config.auth.clientId, authCodeRequest.authority, authCodeRequest.scopes);
-
-            // Handle response from hash string.
-            const result = await interactionHandler.handleCodeResponse(hash, state, authClient.authority, this.networkClient);
+        const popupClient = new PopupClient(this.config, this.browserStorage, this.browserCrypto, this.logger, this.eventHandler, this.navigationClient);
 
+        return popupClient.acquireToken(request).then((result) => {
```
Comment (pkanher617) [inline lib/msal-browser/src/app/ClientApplication.ts:237]: I'm not sure I see the value of mixing .then and async/await, but we can change this later.

**Review by pkanher617 — APPROVED** [review]
Couple things that might need some updates in the future, but looks ok for now. Please make sure tests pass before merging.

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 2 above)
Comment (tnorling) [reply lib/msal-browser/src/app/ClientApplication.ts:237]: Can't do async/await here unfortunately. We made the popup APIs synchronous a while back to enable users to configure popups to be opened synchronously or asynchronously.

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 1 above)
Comment (sameerag) [reply lib/msal-browser/src/app/ClientApplication.ts:25]: Well ... :), don't have anything to add. Lets leave it at that for now but if we have two/three silent clients, I do think names matter.

File: change/@azure-msal-browser-30eb62a8-70f1-4fc5-b9d1-12fd3447960e.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (jasonnutter) [inline change/@azure-msal-browser-30eb62a8-70f1-4fc5-b9d1-12fd3447960e.json:2]: AFAIK this doesn't change the API surface, right?

Comment (tnorling) [reply change/@azure-msal-browser-30eb62a8-70f1-4fc5-b9d1-12fd3447960e.json:2]: Right, this is all internal implementation details. Made it minor just for the sheer volume of changes, acknowledging the small risk that something might break as a result

**Review by sameerag — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**